Add `vitejs workflow using jspm (#16)
jspm/jspm.org: pages/docs/workflows.md

```

For an easier `chomp deno-generate` execution that will first ensure the import map is up to date in the task graph.

## Vitejs

> ES modules are supported directly in all major [browsers](https://caniuse.com/es6-module) now and with [es-modu-shims](https://github.com/guybedford/es-module-shims) you can polyfill the rest of the missing features and ship ES modules with confidence.

Reviewer: It's usually best to avoid terms like "you" and instead use passive voice when possible.

```suggestion
ES modules are supported directly in all major [browsers](https://caniuse.com/es6-module) and with [es-module-shims](https://github.com/guybedford/es-module-shims) import maps and other features can be performantly polyfilled when not supported to allow shipping ES modules with confidence.
```

> [Vite](https://vitejs.dev/) is built heavily on the ES modules which makes using both JSPM and Vite a great combination.

Reviewer: ```suggestion
[Vite](https://vitejs.dev/) relies heavily on ES modules which makes using both JSPM and Vite a great combination.
```

> The dependencies are served from JSPM, and the project is bundled using Vite. We released `vite-plugin-jspm` to enable this workflow. You can download the plugin using:

Reviewer: Best not to use terms like "we" as well when possible:

```suggestion
The dependencies are served from JSPM, and the project is bundled using Vite. This workflow is enabled with `vite-plugin-jspm`, which can be downloaded from npm via:
```

```sh
npm install vite-plugin-jspm --save-dev
```

> `vite` and `vite build` can be used dev-server and production builds respectively. For the dev build, JSPM will inject ES Module Shims to polyfill import maps, while in the production build all modules will be build from the CDN. The plugin takes all the options that are supported with [@jspm/generator](https://github.com/jspm/generator#options).

Reviewer: ```suggestion
`vite` and `vite build` can be used for dev server and production builds respectively. For the dev build, JSPM will inject ES Module Shims to polyfill import maps, while in the production build all modules will be built from the CDN. The plugin takes all the options that are supported with [@jspm/generator](https://github.com/jspm/generator#options).
```

For the production build, it can be useful to disable the `polyfilling` and `preload` options:

vite.config.mjs

```js
import { defineConfig } from "vite";
import jspmPlugin from "vite-plugin-jspm";

export default defineConfig({
  clearScreen: true,
  build: {
    polyfillModulePreload: false,
    polyfillDynamicImport: false,
  }
  plugins: [jspmPlugin()],
});
```
An additional option that you can use is `downloadDeps`, which helps in downloading all the dependencies at build time.

> In considering the `downloadDeps` option, there are a number of factors that should be considered:

Reviewer: Word "considered" shouldn't be repeated:

```suggestion
When using the `downloadDeps` option, there are a number of factors that should be considered:
```

> - Loading modules from the JSPM CDN will results in a shared edge cache with other JSPM users resulting in lower latencies for code delivery over the globally distributed CDN network.

Reviewer: Remove duplicate "result" term. Also we should be clearer that this is a benefit:

```suggestion
- Using external CDN dependencies will likely provide the best latency by utilizing the shared globally distributed CDN network - since the JSPM CDN cache is shared at the edge with other JSPM users this leads to shared latency optimization on edges.
```

```

For an easier `chomp deno-generate` execution that will first ensure the import map is up to date in the task graph.

## Vitejs

> ES-Modules are supported directly in all major [browsers](https://caniuse.com/es6-module) now. And with [es-modu-shims](https://github.com/guybedford/es-module-shims) you can polyfill the rest of the missing features and ship es-modules with confidence.

Reviewer: ```suggestion
ES modules are supported directly in all major [browsers](https://caniuse.com/es6-module) now and with [es-modu-shims](https://github.com/guybedford/es-module-shims) you can polyfill the rest of the missing features and ship ES modules with confidence.
```

> Vitejs is built heavily on the esm modules which makes both jspm and vitejs a perfect combination. 

Reviewer: I believe they just call it `Vite`? Also we should make this a link.

```suggestion
[Vite](https://vitejs.dev/) is built heavily on the ES modules which makes using both JSPM and Vite a great combination. 
```

> The dependencies are served from jspm, and the project is bundled using vitejs. We released `vite-plugin-jspm` to enable this workflow. You can download the plugin using.

Reviewer: ```suggestion
The dependencies are served from JSPM, and the project is bundled using Vite. We released `vite-plugin-jspm` to enable this workflow. You can download the plugin using:
```

```sh
npm install vite-plugin-jspm --save-dev
```

> You can add to the config and disable the `polyfilling` and `preload` options from `vitejs` build. As jspm add `es-module-shims` which takes care of polyfilling.

Reviewer: Perhaps rather start with the next line:

```suggestion
`vite` and `vite build` can be used dev-server and production builds respectively. For the dev build, JSPM will inject ES Module Shims to polyfill import maps, while in the production build all modules will be build from the CDN. The plugin takes all the options that are supported with [@jspm/generator](https://github.com/jspm/generator#options).

For the production build, it can be useful to disable the `polyfilling` and `preload` options:
```

Actually thinking about this though - why aren't these options the default for the build in the JSPM Vite plugin?

vite.config.mjs

```js
import { defineConfig } from "vite";
import jspmPlugin from "vite-plugin-jspm";

export default defineConfig({
  clearScreen: true,
  build: {
    polyfillModulePreload: false,
    polyfillDynamicImport: false,
  }
  plugins: [jspmPlugin()],
});
```

> And you can use `vite` and `vite build` respectively for dev-server and production builds. The plugin takes all the options that are supported with [@jspm/generator](https://github.com/jspm/generator#options).

Reviewer: ```suggestion
```

As mentioned, this should be moved up I think?

> An additational option that you can use is `downloadDeps`, which helps in downloading all the dependencies at build time.

Reviewer: ```suggestion
An additional option that you can use is `downloadDeps`, which helps in downloading all the dependencies at build time.
```

Wasn't this the reason for disabling the polyfill? Or does that apply for build mode regardless?

Author: It wil apply for build mode, the reason we disable polyfill is `es-module-shims` is taking care of those. And no need to force vite to do all those again.

> But loading dependecies has many advantages

Reviewer: ```suggestion
In considering the `downloadDeps` option, there are a number of factors that should be considered:
```

> - Served using globally distributed CDN network.
> - No need to bust cache for the entire build when there is a chagne in project, deps are always cached.
> - Very minimal footprint of the app as the dependencies are handled by the CDN.

Reviewer: ```suggestion
- Loading modules from the JSPM CDN will results in a shared edge cache with other JSPM users resulting in lower latencies for code delivery over the globally distributed CDN network.
- Using import maps in production results in there being no need to cache bust the entire build. When there is a small change in project, unchanged dependencies remain cached.
- Very minimal footprint of the app as the dependencies are handled by the CDN.
```

Reviewer: Perhaps also add:

- Performance may be faster with or without `downloadDeps` depending on the exact loading profile and caching requirements.
- Having all sources collected together via `downloadDeps` can be useful for a fully self-contained distribution.
